### ISSUE TITLE
loss increase (custom dataset)

### ISSUE THREAD

**Issue by ramdhan1989** [XiYe20/VPTR#18]
Hi, 
is there solution or insight for the problem? 

![Image](https://github.com/user-attachments/assets/77c8a6a8-6433-44b9-b3e6-d7c167e854f2)

thank you

**Comment by XiYe20** [comment]
Hi, it is hard for me to help you debug this. My suggestion is to firstly disable GDL training and GAN training, only use MSE firstly. If it works properly, add those loss back, one by one, and tune the weight. Maybe check your training data preprocess too; have a look at the output layer of the autoencoder, make sure its output range can match your training data value range.

**Comment by ramdhan1989** [comment]
Well noted, Thanks I got reasonable result now for autoencoder. However, when training FAR method, I didnt get good result. it looks blury and not moving but the result from autoencoder only looks good. I tried both original loss function and mse only but both methods didnt give good result. do you have any suggestion?

**Comment by XiYe20** [comment]
Hi, you can try many tricks for optimization, for example, to increase the size of the Transformer, different learning rate, etc.. If it still fails to generate good prediction results for FAR method, I suggest you to replace the AutoEncoder with VQVAE/VQGAN, in this case, the distribution of latent codes are discrete Categorical, so we also need to change the loss function for the FAR training. Please see papers like this one: Taming Transformers for High-Resolution Image Synthesis, for more details. It may help for some dataset. If you want to keep the latent code of your custom dataset to be with a Continuous distribution, like Gaussian, you may refer to some recent better techniques: GIVT: Generative Infinite-Vocabulary Transformers, Autoregressive Image Generation without Vector Quantization.
